managed/src/main/java/com/yugabyte/yw/scheduler/Scheduler.java

```diff
@@ -249,28 +249,35 @@ private void runMultiTableBackupsTask(Schedule schedule) {
       taskParams.universeUUID, universe.name);
   }
 
-  private void runDeleteBackupTask(Backup backup, Schedule schedule) {
+  public void runDeleteBackupTask(Backup backup) {
     if (backup.state != Backup.BackupState.Completed) {
       LOG.warn("Cannot delete backup {} since it is not in completed state.",
         backup.backupUUID);
       return;
     }
     BackupTableParams backupParams = Json.fromJson(backup.backupInfo, BackupTableParams.class);
     Universe universe = null;
+    boolean isScheduled = (backup.getScheduleUUID() != null) ? true : false;
     try {
       universe = Universe.get(backupParams.universeUUID);
     } catch (Exception e) {
-      schedule.stopSchedule();
+      if (isScheduled){
+        Schedule schedule = Schedule.get(backup.getScheduleUUID());
+        schedule.stopSchedule();
+      }
       return;
     }
-    UUID customerUUID = schedule.getCustomerUUID();
+    UUID customerUUID = backup.customerUUID;
     Customer customer = Customer.get(customerUUID);
-    JsonNode params = schedule.getTaskParams();
     DeleteBackup.Params taskParams = new DeleteBackup.Params();
     taskParams.customerUUID = customerUUID;
     taskParams.backupUUID = backup.backupUUID;
     UUID taskUUID = commissioner.submit(TaskType.DeleteBackup, taskParams);
-    ScheduleTask.create(taskUUID, schedule.getScheduleUUID());
+    if (isScheduled) {
+      Schedule schedule = Schedule.get(backup.getScheduleUUID());
+      JsonNode params = schedule.getTaskParams();
+      ScheduleTask.create(taskUUID, backup.getScheduleUUID());
```

Comment:
@Arnav15 Remind me why we need to create a ScheduleTask here for deletion? and create customer task as well?

Reply:
We do not need this block as discussed.